mesa_geo/tile_layers.py

```diff
@@ -0,0 +1,74 @@
+"""
+Tile Layers
+-----------
+"""
+
+from __future__ import annotations
+
+import dataclasses
+from dataclasses import dataclass
+from typing import Union
+from enum import Enum
+
+import xyzservices
+
+LeafletOption = Union[str, bool, int, float]
+
+
+@dataclass
+class RasterWebTile:
+    """
+    A class for the background tile layer of Leaflet map that uses a raster
+    tile server as the source of the tiles.
+
+    The available options can be found at: https://leafletjs.com/reference.html#tilelayer
+    """
+
+    url: str
+    options: dict[str, LeafletOption] | None = None
+    kind: str = "raster_web_tile"
+
+    @classmethod
+    def from_xyzservices(cls, provider: xyzservices.TileProvider) -> RasterWebTile:
+        """
+        Create a RasterWebTile from an xyzservices TileProvider.
+
+        :param provider: The xyzservices TileProvider to use.
+        :return: A RasterWebTile instance.
+        """
+        provider_dict = dict(provider)
+        url = provider_dict.pop("url")
+        attribution = provider_dict.pop("html_attribution")
+        provider_dict.update({"attribution": attribution})
+        return cls(url=url, options=provider_dict)
+
+    def to_dict(self) -> dict:
+        return dataclasses.asdict(self)
+
+
+@dataclass
+class WMSWebTile(RasterWebTile):
+    """
+    A class for the background tile layer of Leaflet map that uses a WMS
+    service as the source of the tiles.
+
+    The available options can be found at: https://leafletjs.com/reference.html#tilelayer-wms
+    """
+
+    kind = "wms_web_tile"
+
+
+class MapTiles(Enum):
+    """
+    Pre-configured map tiles for the background tile layer of Leaflet map.
+    """
+
+    OPEN_STREET_MAP = RasterWebTile(
```

Comment:
Doesn't `xyzservices` already have this? https://github.com/geopandas/xyzservices/blob/5390ede63a1b05e9839fb825fc6aca61f2f16df5/xyzservices/data/providers.json#L2-L8

Reply:
Yes, but I preferred to have the default value from our own repo, instead of using that from `xyzservices`.

Shall I remove the `MapTiles` class completely for now?

Reply:
I first need to understand the difference with `xyzservices`'s tiles. If they are the same, I think we shouldn't duplicate it.

Reply:
Well, these are exactly the same : )

Reply:
Ok, I'll remove them then.

Reply:
Updated!